File: backend/src/main/java/com/wildBirds/BlueChat/api/rest/dto/ChannelDto.java
package com.wildBirds.BlueChat.api.rest.dto;

import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class ChannelDto {
    private Long idChannel;
    private String name;
    private boolean isPublic;
    private UserDtoShort channelOwner;
    private List<UserDto> userList = new ArrayList<>();
    private PhotoDto photoDto;

    public boolean getIsPublic() {
        return isPublic;
    }

    public void setIsPublic(boolean isPublic) {
        this.isPublic = isPublic;
    }

}

File: mobile/src/app/repository/channel-repository/channelDto.ts
import {UserDtoShort} from '../user-repository/userDtoShort';
import {PhotoDto} from '../photo-repository/photoDto';

export class ChannelDto {
    private id: number;
    private name: string;
    private isPublic: boolean;
    private channelOwner: UserDtoShort;
    private userList: Array<UserDtoShort>;
    private photo: PhotoDto;
}